hgweb/command-runner.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Define the list of allowed commands
+allowed_commands=("verify")
+
+# Get the project code and command name from the URL
+IFS='/' read -ra PATH_INFO <<< "$PATH_INFO"
+project_code="${PATH_INFO[1]}"
+command_name="${PATH_INFO[2]}"
+
+# Ensure the project code and command name are safe to use in a shell command
+if [[ ! $project_code =~ ^[a-z0-9-]+$ ]] || [[ ! $command_name =~ ^[a-zA-Z0-9]+$ ]]; then
```

Comment:
I thought quotes would be required around project_code and command_name to avoid issues if someone puts %20 into the URL and it decodes into a space character, but I checked the Bash documentation (and tested it myself) and Bash treats `[[` expressions specially, and does not split expressions on spaces. So this line is fine as-is, contrary to my first impression.